graph/iid/IID.java

```diff
@@ -46,12 +46,12 @@ public boolean equals(Object object) {
         if (this == object) return true;
         if (object == null || getClass() != object.getClass()) return false;
         IID that = (IID) object;
-        return Arrays.equals(this.bytes, that.bytes);
+        return this.bytes.equals(that.bytes);
     }
 
     @Override
     public final int hashCode() {
-        if (hash == 0) hash = Arrays.hashCode(bytes);
+        if (hash == 0) hash = bytes.hashCode();
```

Comment:
this is the primary user of bytes hashcode - which is why `ByteArray` itself isn't caching the hashcode right now (just uses more memory)

Reply:
ok nvm this was actualy worthwhile according to benchmark